torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2003,6 +2012,245 @@ def clip_grad_norm_(
                 assert p.grad is not None
                 p.grad.detach().mul_(clip_coef.to(p.grad.device))
 
+    @staticmethod
+    def full_optim_state_dict(
+        model: torch.nn.Module,
+        optim: torch.optim.Optimizer,
+        optim_input: Optional[Union[
+            List[Dict[str, Any]], Iterable[torch.nn.Parameter],
+        ]] = None,
+    ) -> Dict[str, Any]:
+        """
+        Consolidates the full optimizer state on the target rank and returns it
+        as a :class:`dict` following the convention of normal
```

Comment:
To-do: Update these comments since we no longer match local equivalent in all cases.